plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp

```diff
@@ -1069,18 +1080,65 @@ void SelectionUndoHelper::doIt()
             m_insertedRefs.size(),
             m_removedRefs.size());
     m_proxy->m_pleaseIgnoreSelection = true;
-    m_modifier1.doIt();
-    m_modifier2.doIt();
-    m_proxy->insertTransformRefs(m_insertedRefs, nodes::ProxyShape::kSelection);
-    m_proxy->removeTransformRefs(m_removedRefs, nodes::ProxyShape::kSelection);
-    m_proxy->selectedPaths() = m_paths;
-    if (!m_internal) {
-        MGlobal::setActiveSelectionList(m_newSelection, MGlobal::kReplaceList);
+    if (!MayaUsdProxyShapePlugin::useVP2_NativeUSD_Rendering()) {
+        m_modifier1.doIt();
+        m_modifier2.doIt();
+        m_proxy->insertTransformRefs(m_insertedRefs, nodes::ProxyShape::kSelection);
+        m_proxy->removeTransformRefs(m_removedRefs, nodes::ProxyShape::kSelection);
+        m_proxy->selectedPaths() = m_paths;
+        if (!m_internal) {
+            MGlobal::setActiveSelectionList(m_newSelection, MGlobal::kReplaceList);
+        }
+        m_proxy->m_pleaseIgnoreSelection = false;
+        if (!MGlobal::optionVarIntValue("AL_usdmaya_ignoreLockPrims")) {
+            m_proxy->constructLockPrims();
+        }
     }
-    m_proxy->m_pleaseIgnoreSelection = false;
-    if (!MGlobal::optionVarIntValue("AL_usdmaya_ignoreLockPrims")) {
-        m_proxy->constructLockPrims();
+#if defined(WANT_UFE_BUILD)
+    else {
+        m_proxy->m_selectedPaths.clear();
+        for (auto& it : m_newUFESelection) {
+            const auto& path = it->path();
+            const auto& pathStr = path.string();
+            const auto  index = pathStr.find_first_of('/');
```

Comment:
Would be more effective to only deal with the USD segment, instead of hunting for the '/' character in a string version of the path.